pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -726,12 +730,29 @@ func getOSDInfo(d *apps.Deployment) ([]OSDInfo, error) {
 		}
 		locationPrefix := "--crush-location="
 		if strings.HasPrefix(a, locationPrefix) {
+			locationFound = true
 			// Extract the same CRUSH location as originally determined by the OSD prepare pod
 			// by cutting off the prefix: --crush-location=
 			osd.Location = a[len(locationPrefix):]
 		}
 	}
 
+	if !locationFound {
+		pods, err := c.context.Clientset.CoreV1().Pods(d.Namespace).List(metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%d", OsdIdLabelKey, osd.ID)})
+		if err != nil {
+			logger.Errorf("error getting pods %v", err)
+		}
+		nodeName := pods.Items[0].Spec.NodeName
+		hostName, err := k8sutil.GetNodeHostName(c.context.Clientset, nodeName)
+		if err != nil {
+			logger.Errorf("error getting hostname %v", err)
+		}
+		if d.GetLabels()[portableKey] == "true" {
```

Comment:
Wait, you want to set the pvc name as the hostName if `portable == "true"`, right? So it should be something like:
```
val, ok := d.GetLabels()[portableKey]
if ok && val == "true" {
  hostName = pvcName // should be able to get the pvc name from a label on the osd pod
}
// no "else" block is needed since the hostName is already set as needed for non-portable osds
```